src/sync/SyncQueue.js

```diff
@@ -55,36 +55,52 @@ export class SyncQueue {
       // If a delete, first remove any sync out records that already have the id,
       // so that sync doesn't try to refer to them next time it does a push
       if (changeType === DELETE) {
-        const recordsToDelete = this.database.objects('SyncOut')
-                                             .filtered('recordId == $0', record.id);
+        const recordsToDelete = this.database
+          .objects('SyncOut')
+          .filtered('recordId == $0', record.id);
         this.database.delete('SyncOut', recordsToDelete);
       }
       switch (changeType) {
         case CREATE:
         case UPDATE:
         case DELETE: {
+          let duplicate = false;
           if (!record.id) return;
-          const duplicate = this.database.objects('SyncOut')
-                              .filtered(
-                                'changeType == $0 && recordType == $1 && recordId == $2',
-                                changeType,
-                                recordType,
-                                record.id)
-                              .length > 0;
+
+          if (recordType === 'Transaction' && record.isFinalised) {
+            // Remove old snapshots from the same transaction to be sync with lines
+            const result = this.database
+              .objects('SyncOut')
+              .filtered('recordType == $0 && recordId == $1', recordType, record.id);
+
+            Object.entries(result).forEach(([key, value]) => {
+              this.database.delete('SyncOut', value);
```

Comment:
Cool! Chris explained here to me, I see what you mean know. If that way the record will go to the end of the line like that is fine. We just also have to search for `recordType == CREATE` records also when we update the `changeTime`.